public/app/plugins/datasource/cloudwatch/query-runner/CloudWatchRequest.ts

```diff
@@ -50,13 +50,18 @@ export abstract class CloudWatchRequest {
         return { ...result, [key]: null };
       }
 
-      const newValues = this.expandVariableToArray(value, scopedVars);
+      const newValues = this.expandVariableToArrayIfTemplated(value, scopedVars);
       return { ...result, [key]: newValues };
     }, {});
   }
 
   // get the value for a given template variable
-  expandVariableToArray(value: string, scopedVars: ScopedVars): string[] {
+  expandVariableToArrayIfTemplated(value: string, scopedVars: ScopedVars): string | string[] {
+    // If no template variables, simply return the value.
+    if (!this.templateSrv.getVariables.length) {
```

Comment:
`getVariables` will return variables also for other data sources in the current dashboard, so think you can remove this if statement. What you can instead is to simply return the string on line 73 rather than putting it in an array. If we reach line 73, we know it's not a variable.

Reply:
Yeah I had that originally but wasn't sure if it could reach that case otherwise. Will amend 👍